File: MemeStream/01-RestfulAPI/src/models/Meme.js
import mongoose from 'mongoose';

const memeSchema = new mongoose.Schema(
  {
    userId: { type: String, required: true, index: true },
    title: { type: String, required: true, trim: true },
    imageUrl: { type: String, required: true },
    caption: { type: String, default: '' },
    tags: { type: [String], default: [] },
    source: { type: String, default: '' }
  },
  { timestamps: true }
);

export default mongoose.model('Meme', memeSchema);

File: Android/app/src/main/java/com/altaaf/memeice/model/Meme.kt
package com.altaaf.memeice.model

data class Meme(
    val _id: String,
    val userId: String,
    val title: String,
    val imageUrl: String,
    val caption: String?,
    val tags: List<String>?,
    val source: String?,
    val createdAt: String?,
    val updatedAt: String?
)